eb_hooks.py

```diff
@@ -684,7 +684,13 @@ def post_sanitycheck_cuda(self, *args, **kwargs):
     Remove files from CUDA installation that we are not allowed to ship,
     and replace them with a symlink to a corresponding installation under host_injections.
     """
-    if self.name == 'CUDA':
+    # Make sure we only do this for CUDA and only if we are doing a CVMFS installation
+    is_eessi_install = (
+        self.installdir.startswith("/cvmfs/software.eessi.io/versions")
+        and not build_option("sanity_check_only")
```

Comment:
Pfft, not a big fan of this.

Why isn't this done through a post-install hook, is there a reason for that?

Also not a fan of the hardcoded check for `/cvmfs/software.eessi.io`... What if we do a "test" install of CUDA in `dev.eessi.io`?

We want to try and make sure we never ship a CUDA installation that isn't stripped down?

Reply:
Suggestions welcome. Hook is where it is so that it doesn't impact the installation itself including the sanity check

Reply:
Obviously I can add an alternate check for dev.eessi.io but I'm sensing general negativity

Reply:
1) W.r.t. this being a post-sanity-check hook: it's both good and bad this it's done there: good because it only kicks in after the sanity check performed by EasyBuild, but bad because we fiddle with the installation after the sanity check has completed (that's a necessary evil I guess).
In general, I'm fairly OK with that, but the additional checks being added now increase the risk that we accidentally won't strip the CUDA installation, imho.

I think we can probably guarantee that the full CUDA installation in `host_injections` will be there when we install the CUDA for the software layer that will be stripped down, so with that in mind it would be better to move this stripping down action to a post-install hook.

2) I'm not sure I fully understand the need for the check "are we installing under EESSI", but rather than having a very strict check like "install path starts with `/cvmfs/software.eessi.io/versions` or `/cvmfs/dev.eessi.io/...`, we could make it a regex check which is a lot less likely to accidentally pass when we don't want it to (like check for `^/cvmfs/[^/]*.eessi.io/`).
Can you clarify why we need that though? You basically want to make sure that this bit isn't triggered when we do the full CUDA install in `host_injections`?
Then why not explicitly check for that (if `/host_injections/` in installpath => skip stripping)?

3) Fixing the CPU duplication issue would be nice, but it's not something we should pursue right now I think, that could be a goal for the next EESSI version (since it's basically just cleanup, no real other impact)?

Reply:
Checking for `host_injections` is not enough when deciding when to trigger the hook, people may be using the hook via `EESSI-extend` in which case you don't it triggered

Reply:
The regex with `versions` should do it

Reply:
I've done an implementation in https://github.com/EESSI/software-layer/pull/735/commits/a3eddb49ef404fa727436f64e59afb68e33d80f6 but now the method contains two hard-coded patterns: one to match to an EESSI-distributed repo and one for the location of `host_injections`, we may want to move these to a global definition